paymentsheet/src/main/java/com/stripe/android/link/ui/verification/VerificationBody.kt

```diff
@@ -0,0 +1,338 @@
+
+        Spacer(modifier = Modifier.size(8.dp))
+
+        Text(
+            text = stringResource(R.string.stripe_link_verification_message_short, state.redactedPhoneNumber),
+            modifier = Modifier
+                .testTag(VERIFICATION_SUBTITLE_TAG)
+                .fillMaxWidth(),
+            textAlign = TextAlign.Companion.Center,
+            style = LinkTheme.typography.body,
+            color = LinkTheme.colors.textSecondary
+        )
+
+        Spacer(modifier = Modifier.size(24.dp))
+
+        StripeThemeForLink {
+            OTPElementUI(
+                enabled = !state.isProcessing,
+                element = otpElement,
+                middleSpacing = 8.dp,
+                boxSpacing = 8.dp,
+                otpInputPlaceholder = " ",
+                boxShape = LinkTheme.shapes.large,
+                modifier = Modifier
+                    // 48dp per OTP box plus 8dp per space
+                    .width(328.dp)
+                    .testTag(VERIFICATION_OTP_TAG),
+                colors = OTPElementColors(
+                    selectedBorder = LinkTheme.colors.borderSelected,
+                    placeholder = LinkTheme.colors.textPrimary,
+                    background = LinkTheme.colors.surfaceSecondary
+                ),
+                focusRequester = focusRequester
+            )
+        }
+
+        AnimatedVisibility(visible = state.errorMessage != null) {
+            ErrorText(
+                text = state.errorMessage?.resolve(LocalContext.current).orEmpty(),
+                modifier = Modifier
+                    .padding(top = 16.dp)
+                    .testTag(VERIFICATION_ERROR_TAG)
+                    .fillMaxWidth()
+            )
+        }
+
+        Spacer(modifier = Modifier.size(36.dp))
+        ResendCodeButton(
+            isProcessing = state.isProcessing,
+            isSendingNewCode = state.isSendingNewCode,
+            onClick = onResendCodeClick,
+        )
+
+        if (state.isDialog.not()) {
+            Spacer(modifier = Modifier.size(24.dp))
+            ChangeEmailRow(
+                email = state.email,
+                isProcessing = state.isProcessing,
+                onChangeEmailClick = onChangeEmailClick,
+            )
+        }
+
+        Spacer(modifier = Modifier.size(12.dp))
+    }
+}
+
+/**
+ * A wrapper for the content of the verification screen.
+ *
+ * @param isDialog whether the screen is displayed as a dialog or not
+ * @param content the content to be displayed
+ */
+@Composable
+private fun ContentWrapper(
+    isDialog: Boolean,
+    onBackClicked: () -> Unit,
+    content: @Composable ColumnScope.() -> Unit
+) {
+    if (isDialog) {
+        Box {
+            IconButton(
+                modifier = Modifier
+                    // - IconButton ensures a 48.dp touch target for accessibility targets.
+                    // - The dialog padding is 24.dp.
+                    // - The icon is 16.dp
+                    .padding(12.dp)
```

Comment:
small calculation to ensure we match design paddings keeping the button accessible.